include/zephyr/rtio/rtio.h

```diff
@@ -1674,6 +1713,91 @@ static inline int z_impl_rtio_submit(struct rtio *r, uint32_t wait_count)
 }
 #endif /* CONFIG_RTIO_SUBMIT_SEM */
 
+/*
+ * @brief Create a chain of SQEs representing a bus transaction to read a reg.
+ *
+ * The RTIO-enabled bus driver is isnstrumented to perform bus read ops
+ * for each register in the list.
+ *
+ * Usage:
+ *
+ * uint8_t regs_list[]  = { reg_addr1, reg_addr2. ... }
+ * struct rtio_reg_buf rbuf[] = {{mem_addr_1, mem_len_1},
+ *                               {mem_addr_2, mem_len_2},
+ *                               ...
+ *                              };
+ *
+ * rtio_read_transaction(dev,
+ *                       regs_list,
+ *                       ARRAY_SIZE(regs_list),
+ *                       BUS_SPI,
+ *                       rbuf,
+ *                       sqe,
+ *                       iodev,
+ *                       rtio,
+ *                       op_cb);
+ *
+ * @param rtio_bus_type bus type (i2c, spi, i3c)
+ * @param rtio_reg_buf buffer of memory chunks
+ * @param iodev_sqe IODEV submission for the await op
+ * @param iodev IO device
+ * @param r RTIO context
+ * @param rtio_callback_t callback routine at the end of op
+ */
+static inline void rtio_read_transaction(struct rtio *r,
+					 struct rtio_iodev *iodev,
+					 uint8_t *regs,
+					 uint8_t regs_num,
+					 rtio_bus_type bus_type,
+					 struct rtio_reg_buf *buf,
+					 struct rtio_iodev_sqe *iodev_sqe,
+					 const struct device *dev,
+					 rtio_callback_t complete_op_cb)
+{
+	struct rtio_sqe *write_addr;
+	struct rtio_sqe *read_reg;
+	struct rtio_sqe *complete_op;
+	uint8_t i;
+
+	for (i = 0; i < regs_num; i++) {
+
+		write_addr = rtio_sqe_acquire(r);
+		read_reg = rtio_sqe_acquire(r);
+
+		if (write_addr == NULL || read_reg == NULL) {
+			rtio_iodev_sqe_err(iodev_sqe, -ENOMEM);
+			rtio_sqe_drop_all(r);
+			return;
+		}
+
+		if (rtio_is_spi(bus_type)) {
+			regs[i] |= 0x80; /* mark the SPI read transaction */
```

Comment:
I feel like this should be configurable: there are some drivers that have this inverted (as in: 1 when writing, 0 when reading).

Here's one in-tree example: https://github.com/zephyrproject-rtos/zephyr/blob/4ec435165753bf353b35ae700c3ed5a1421eb46b/drivers/sensor/pixart/pat9136/pat9136_reg.h#L11-L12

Reply:
I see. Yes, I remember now that you mentioned it during the WG call, but at the end I forgot it. So, the best thing to do is that the driver would raise the proper bit in ```regs_list[].reg_addr``` (regs_list is the second argument)